src/Command/OgAddFieldCommand.php

```diff
@@ -0,0 +1,146 @@
+        $this->getOgFields()
+      );
+      $input->setOption('field_id', $field_id);
+    }
+
+    if (!$field_name = $input->getOption('field_name')) {
+      $field_name = $io->ask($this->getDefinition()->getOption('field_name')->getDescription());
+      $input->setOption('field_name', $field_name);
+    }
+
+    if (!$entity_type = $input->getOption('entity_type')) {
+      $entity_type = $io->choiceNoList(
+        $this->getDefinition()->getOption('entity_type')->getDescription(),
+        $this->getEntityTypes()
+      );
+      $input->setOption('entity_type', $entity_type);
+    }
+
+    if (!$input->getOption('bundle')) {
+      $bundles = \Drupal::service('entity_type.bundle.info')->getBundleInfo($entity_type);
+      $input->setOption('bundle', $io->choiceNoList(
+        $this->getDefinition()->getOption('bundle')->getDescription(),
+        array_keys($bundles)
+      ));
+    }
+
+    if ($this->fieldIsAudienceField($field_id) && !$input->getOption('target_entity')) {
+      $input->setOption('target_entity', $io->choiceNoList(
+        $this->getDefinition()->getOption('target_entity')->getDescription(),
+        $this->getEntityTypes()
+      ));
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function execute(InputInterface $input, OutputInterface $output) {
+    $io = new DrupalStyle($input, $output);
+
+    $settings = [
+      'field_name' => $input->getOption('field_name'),
+    ];
+
+    if ($this->fieldIsAudienceField($input->getOption('field_id'))) {
+      $settings['field_storage_config']['settings'] = [
+        'target_type' => $input->getOption('target_entity'),
+      ];
+    }
+
+    Og::createField($input->getOption('field_id'), $input->getOption('entity_type'), $input->getOption('bundle'), $settings);
+    $io->info('The field attached successfully.');
+  }
+
+  /**
+   * Get OG fields.
+   *
+   * @return array
+   *   List of OG fields ID.
+   */
+  protected function getOgFields() {
+    return array_map(function ($item) {
+      return $item['id'];
+    },
+    $this->getOgPluginManager()->getDefinitions());
+  }
+
+  /**
+   * Get all the entity types.
+   *
+   * @return array
+   *   List of entity IDs.
+   */
+  protected function getEntityTypes() {
+    $groups = \Drupal::service('entity_type.repository')->getEntityTypeLabels(TRUE);
+    return array_map(function ($item) {
+      return $item->render();
+    }, $groups['Content']);
+  }
+
+  /**
+   * Return OG plugin manager.
+   *
+   * @return \Drupal\og\OgFieldsPluginManager
+   *   OG plugin manager instance.
+   */
+  protected function getOgPluginManager() {
+    return \Drupal::getContainer()->get('plugin.manager.og.fields');
+  }
+
+  /**
+   * Check if the field is an audience field or not.
+   *
+   * @param string $field_id
+   *   The field ID.
+   *
+   * @return bool
+   *   True or false if the field is an audience or not.
+   */
+  protected function fieldIsAudienceField($field_id) {
```

Comment:
I don't think you need specific check for audience field. you just need a check if it's an ER field